### PR TITLE
Introduce Golang-based benchmarking utility

### PR DESCRIPTION
**Pull request by edigaryev**
Resolves https://github.com/cirruslabs/tart/issues/762.

### CODE REVIEW

File: benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Benchmark
```
Comment (fkorotkov) [inline benchmark/README.md:1]: Maybe run it on the M2 Pro `dev-mini` host we have a include the run information at the end of readme? Including host and guest macOS versions (should be 14.4.1).

Comment (edigaryev) [reply benchmark/README.md:1]: Added in https://github.com/cirruslabs/tart/pull/769/commits/f0adb8b016a09eba67f185ae639888683824e223.

File: benchmark/internal/executor/tart/tart.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+	if err := Cmd(ctx, tart.logger, "clone", baseImage, tart.vmName); err != nil {
+		return nil, err
+	}
+
+	vmRunCtx, vmRunCancel := context.WithCancel(ctx)
+	tart.vmRunCancel = vmRunCancel
+
+	go func() {
+		_ = Cmd(vmRunCtx, tart.logger, "run", "--no-graphics", tart.vmName)
```
Comment (fkorotkov) [inline benchmark/internal/executor/tart/tart.go:44]: Do you think it makes sence to match guest CPU/memory with the host. I guess for the current IO testing it won't make a difference but in the future we'll need to have such arguments.

Comment (edigaryev) [reply benchmark/internal/executor/tart/tart.go:44]: I think this totally depends on what we're trying to benchmark for. For now the defaults look reasonable to me.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**